dio/lib/src/dio/dio_for_browser.dart

```diff
@@ -29,9 +27,31 @@ class DioForBrowser with DioMixin implements Dio {
     String lengthHeader = Headers.contentLengthHeader,
     Object? data,
     Options? options,
-  }) {
-    throw UnsupportedError(
-      'The download method is not available in the Web environment.',
+  }) async {
+    options ??= DioMixin.checkOptions('GET', options);
+
+    options = options.copyWith(responseType: ResponseType.bytes);
+
+    // Set receiveTimeout to 48 hours because `Duration.zero` not work!
+    options = options.copyWith(receiveTimeout: const Duration(hours: 48));
+
+    final Response response = await request(
+      urlPath,
+      data: data,
+      options: options,
+      queryParameters: queryParameters,
+      cancelToken: cancelToken,
+      onReceiveProgress: onReceiveProgress,
     );
+
+    final completer = Completer<Response>();
+
+    // Create blob url from byte data
+    response.data = Url.createObjectUrlFromBlob(Blob([response.data]));
+
+    // Set response in Completer
+    completer.complete(response);
+
+    return DioMixin.listenCancelForAsyncTask(cancelToken, completer.future);
```

Comment:
Why do we need extra handling with the CancelToken? Isn't it already handled in the adapter?

Reply:
I did this based on the [`dio_for_native`](https://github.com/cfug/dio/blob/6e5a96b957c302bc1010edb78b0f64d204dea303/dio/lib/src/dio/dio_for_native.dart#L188) file.

Reply:
It uses the explicitly handling because there are subscriptions that need to be cancelled which seems not related with the browser implementation.

Reply:
What is your suggestion?

Reply:
The `completer` in the `dio_for_native.dart` handles data and errors which cover all cases, in which the current implementation does not look the same. So they should be at least the same behavior.

Reply:
What should I use instead of  `return DioMixin.listenCancelForAsyncTask(cancelToken, completer.future);`?

Reply:
It is not about the `listenCancelForAsyncTask`, the completer does not handle any errors which it should.

https://github.com/cfug/dio/blob/47abf18387e89983f010771e19e6d643f5d043b6/dio/lib/src/dio/dio_for_native.dart#L133-L183

Reply:
Changes were made